psp34_shiden/contracts/shiden34/src/lib.rs

```diff
@@ -134,7 +173,7 @@ pub mod shiden34 {
         /// Mint next available token for the caller
         #[ink(message, payable)]
         fn mint_next(&mut self) -> Result<(), PSP34Error> {
-            self.check_value(self.env().transferred_value(), 1)?;
+            self._check_value(self.env().transferred_value(), 1)?;
```

Comment:
You basically only check overflow here (as you return Result<()>
if `mint_amount` * price per mint doesn't overflow
`transfered_value` will be equal to `value`

```
  if (mint_amount as u128).overflowing_mul(self.price_per_mint).1 {
      return SP34Error::Custom("BadMintValue".to_string())
  }
```

Reply:
sorry, I don't get this comment. I'll move this comment to new PR